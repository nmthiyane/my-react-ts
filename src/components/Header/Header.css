.App-logo-active {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-logo {
  height: 80px;
}

.App-header {
  background-color: #014373;
  height: 160px;
  padding: 20px;
  color: white;
}

.App-title {
  margin-top: 10px;
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
