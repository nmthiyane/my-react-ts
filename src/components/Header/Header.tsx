import * as React from 'react';
import './Header.css';

import { ScoreBoard } from '../';

interface Props {
	raceInProgress: boolean;
	lastWinner: string;
}

export class Header extends React.Component<Props, {}> {
	render() {
		const { lastWinner, raceInProgress } = this.props;

		return (
			<header className="App-header" >
				<img
					src={require('../../images/logo.svg')}
					className={raceInProgress ? 'App-logo-active' : 'App-logo'}
					alt="logo"
				/>
				<h1 className="App-title">Implementation Race</h1>
				<ScoreBoard lastWinner={lastWinner} />
			</header>
		);
	}
}