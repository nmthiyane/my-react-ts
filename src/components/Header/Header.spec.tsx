import * as React  from 'react';
import { Header } from './Header';
const renderer  = require('react-test-renderer');

describe('Snapshot test for the entire app', () => {
	let lastWinner: string =  'nmthiyane';
	let raceInProgress: boolean = false;
	it('renders a snapshot', () => {
	const tree = renderer.create((
		<Header 
			lastWinner={lastWinner} 
			raceInProgress={raceInProgress}
		/>
		)).toJSON();
	expect(tree).toMatchSnapshot();
	});
});