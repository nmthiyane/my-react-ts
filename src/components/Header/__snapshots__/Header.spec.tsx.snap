// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the entire app renders a snapshot 1`] = `
<header
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src=""
  />
  <h1
    className="App-title"
  >
    Implementation Race
  </h1>
  <div
    className="ScoreBoard"
  >
    <h2
      className="winner"
    >
      Winner:Â 
      <span>
        nmthiyane
      </span>
    </h2>
  </div>
</header>
`;
